src/lint/collect-grammar-diagnostics.ts

```diff
@@ -97,6 +104,7 @@ export function collectGrammarDiagnostics(
   // *******************
   // Check that SDOs and Early Errors are defined in terms of productions which actually exist
   // Also filter out any "unused parameter" warnings for grammar productions for which the parameter is used in an early error or SDO
+  const ruleId = 'undefined-nonterminal';
```

Comment:
There's actually four distinct errors in this section:

- when the LHS has no corresponding definition
- when the LHS exists but none of its productions match the RHS given in the SDO
- when the RHS contains a "[no LineTerminator here]" restriction
- when the production on the RHS is gated on a parameter (as in `[+a] Whatever`)

They should probably have different IDs. (They already have different messages.)

Reply:
Changed rule IDs for the latter two, but there were no tests to update since these were untested.